src/libos/src/net/socket/unix_socket/unix_addr.rs

```diff
@@ -0,0 +1,150 @@
+use super::*;
+use std::path::{Path, PathBuf};
+use std::{cmp, mem, slice, str};
+
+const MAX_PATH_LEN: usize = 108;
+const SUN_FAMILY_LEN: usize = mem::size_of::<libc::sa_family_t>();
+lazy_static! {
+    static ref SUN_PATH_OFFSET: usize = {
+        let addr = unsafe { mem::MaybeUninit::<libc::sockaddr_un>::uninit().assume_init() };
```

Comment:
I am not really sure that the compiler is smart enough to know that it does not need to allocate memory for `libc::sockaddr_un`.

I think it is better to use this [crate ](https://crates.io/crates/memoffset) to calculate the field offsets.

Reply:
Honestly, that crate also uses `MaybeUninit`, but I used the crate to keep the code clean.